enforcer-parent/commons/src/main/java/org/wso2/choreo/connect/enforcer/commons/logging/ErrorLogPatternConverter.java

```diff
@@ -0,0 +1,83 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.enforcer.commons.logging;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.impl.Log4jLogEvent;
+import org.apache.logging.log4j.core.pattern.ConverterKeys;
+import org.apache.logging.log4j.core.pattern.LogEventPatternConverter;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+
+import java.util.Arrays;
+
+/**
+ * This will create a PatternConverter which is used by log4j PatternConverter
+ * to format the log event into printable format.
+ * %errorDetails can be used in pattern of PatternLayout to get error info to the Log Statements.
+ */
+
+@Plugin(name = "ErrorLogPatternConverter", category = "Converter")
+@ConverterKeys({"errorDetails"})
+public class ErrorLogPatternConverter extends LogEventPatternConverter {
+    protected ErrorLogPatternConverter(String name, String style) {
+        super(name, style);
+    }
+
+    public ErrorLogPatternConverter(String[] options) {
+        super("errorDetails", "errorDetails");
+    }
+
+    public static ErrorLogPatternConverter newInstance(final String[] options) {
+        return new ErrorLogPatternConverter(options);
+    }
+
+    @Override
+    public void format(LogEvent event, StringBuilder toAppendTo) {
+
+        if (event.getClass() == Log4jLogEvent.class) {
+            Log4jLogEvent logEvent = (Log4jLogEvent) event;
+            if (event.getLevel() == Level.ERROR) {
+                if (logEvent.getMessage().getClass() == ParameterizedMessage.class) {
+                    Object[] parameters = ((ParameterizedMessage) logEvent.getMessage()).getParameters();
+                    if (Arrays.stream(parameters).anyMatch(p ->
+                            p.getClass().getName().equals(ErrorLog.class.getName()))) {
+                        Arrays.stream(parameters)
+                                .filter(p -> p.getClass().getName().equals(ErrorLog.class.getName())).forEach((c) -> {
+                                    ErrorLog errorLog = (ErrorLog) c;
+                                    toAppendTo.append("severity:" + errorLog.getSeverity());
+                                    toAppendTo.append(" ");
+                                    toAppendTo.append("code:" + errorLog.getCode());
+                                });
+                    } else {
+                        toAppendTo.append("severity:" + LoggingConstants.Severity.DEFAULT);
+                        toAppendTo.append(" ");
+                        toAppendTo.append("code:" + 0);
+                    }
+                } else {
+                    toAppendTo.append("severity:" + LoggingConstants.Severity.DEFAULT);
+                    toAppendTo.append(" ");
+                    toAppendTo.append("code:" + 0);
+                }
+            }
```

Comment:
can use a logging util method

Reply:
Can't use a util method to takeover the logic here and in CustomJsonLayout, because even the logic in if statements appears same, the code inside if statements is different in two places.